python_modules/dagster/dagster/_core/execution/backfill.py

```diff
@@ -416,28 +427,59 @@ def with_asset_backfill_data(
             asset_backfill_data=(asset_backfill_data if not is_backcompat else None),
         )
 
+    @public
     @classmethod
     def from_asset_partitions(
         cls,
-        backfill_id: str,
         asset_graph: BaseAssetGraph,
-        partition_names: Optional[Sequence[str]],
         asset_selection: Sequence[AssetKey],
-        backfill_timestamp: float,
-        tags: Mapping[str, str],
+        # TODO(deepyaman): Expose `dagster_instance` instead for the public API.
```

Comment:
Should this be done? If that's the case, would have something like `PartitionBackfill.from_asset_partitions()` that takes a `DagsterInstance` and, under the hood, calls `PartitionBackfill._from_asset_partition()` that takes a `DynamicPartitionsStore`. All of the existing implementations could call the internal method (and not be limited to the slightly more narrow `DagsterInstance` type).

The reason for doing this would be that `DynamicPartitionsStore` is not part of the public API. That said, I wanted to check before doing this refactoring.